services/wiki/wiki.go

```diff
@@ -81,6 +87,19 @@ func InitWiki(repo *models.Repository) error {
 	return nil
 }
 
+func checkFileExistence(gitRepo *git.Repository, filePath string) (bool, error) {
```

Comment:
repeatedly running this is very expensive. gitRepo.LsFiles(...) is not free.

Why don't we do somethings else:

* Pass in the raw unchanged name 
* Add ".md" suffix if not there.
* Look for this raw suffixed  name within the stringslice  - that works return true, rawSuffixed, nil
* If not normalize and look for that - if that works returns true, normalized path, nil
* If not return false, raw suffixed, nil

Reply:
Hi and I updated my patch, added a `prepareWikiFileName()` function that did what you described, but I'm not sure if I understood you correctly.

Yeah `gitRepo.LsFiles(...)` is not free, but adding that function seems could only make the code looks cleaner, it still requires two `LsFiles()` calls to look for the file inside that function, or is there another method that better suitable for this use case?

Reply:
No, this wasn't what I was meaning.

You look for two filenames: The one passed in and the normalized path.

```go
func checkFileExistence(gitRepo *git.Repository, filePath, alternate string) (bool, string, error) {
filesInIndex, err := gitRepo.LsFiles(filePath)
	if err != nil {
		log.Error("%v", err)
		return false, err
	}
	if util.IsStringInSlice(filePath, filesInIndex) {
		return true, filePath, nil
	}
	if util.IsStringInSlice(alternate, filesInIndex) {
		return true, alternate, nil
	}
  return false, "", nil
}
```

Although you should be a bit more efficient.

Reply:
Oh, I thought `LsFiles()` only check if the passed file exists. Sorry I don't code in Go too much and I should read the docs first XD

Thanks for your guidance!